helm/resource_release.go

```diff
@@ -269,11 +269,21 @@ func prepareTillerForNewRelease(d *schema.ResourceData, c helm.Interface, name s
 }
 
 func resourceDiff(d *schema.ResourceDiff, meta interface{}) error {
+
+	// Always set desired state to be DEPLOYED
+	err := d.SetNew("status", release.Status_DEPLOYED.String())
+	if err != nil {
+		return err
+	}
+
+	// Get Chart metadata, if we fail - we're done
 	c, _, err := getChart(d, meta.(*Meta))
-	if err == nil {
-		return d.SetNew("version", c.Metadata.Version)
+	if err != nil {
+		return nil
```

Comment:
@stepanstipl Could you please remind - why did you make the call of `getChart` tolerant to errors?

As I see, if the helm client (library) fails to fetch the chart, that should be a valid point for throwing an error, like it happens with other places when `getChart` is called (see `resourceReleaseCreate`, `resourceReleaseUpdate` functions). Otherwise we can miss the diff and the update won't be triggered.

Reply:
Hah, it took me a while to remember/re-figure out why I did that, long time :).

So the `CustomizeDiff`, and therefore `resourceDiff`, is executed before the chart is available in some cases. In that scenario `getChart` would fail and therefore fail the whole `plan`/`apply` run, whil in fact the run would complete just fine.

Good example of this is the `TestAccResourceRelease_repository` test case, when the chart is only available after the repo is added via the `helm_repository` resource.

Reply:
@stepanstipl Thank you for the explanation! Now I understand. Your test case failed because `CustomizeDiff` functions of all resources are called before these resources are created/updated.
I asked just because I run into the opposite situation - my custom repo was unavailable on the network level, but the terraform apply didn't show any diff or error, when it was expected.

It seems that downloading the chart on the diff evaluation stage is not a good idea and we need to find a way how to store everything needed in the terraform state. 
/cc @rporres

Reply:
> but the terraform apply didn't show any diff or error, when it was expected

@legal90 do you mean that even running the `apply` step, with chart trying to install unavailable repo, did not fail? I would expect the TF to fail there.

Reply:
@stepanstipl My case was pretty dummy - the repository URL was unavailable and the HTTP connection to it was timing with a very long delay (~3-4 minutes), which just caused terraform apply to run very long without any error thrown.
Though, there were no diff expected in my case. You're right - if there would be a diff, then terraforn eventually throws an error because it downloads the chart once again in `resourceReleaseUpdate`.